index.d.ts

```diff
@@ -1,288 +1,523 @@
+    manualExpandedKey: IdType<D>
+  }>
 
-  export interface Hooks {
-    columnsBeforeHeaderGroups: any[]
-    columnsBeforeHeaderGroupsDeps: any[]
-    useMain: any[]
-    useColumns: any[]
-    useHeaders: any[]
-    useHeaderGroups: any[]
-    useRows: any[]
-    prepareRow: any[]
-    getTableProps: any[]
-    getRowProps: any[]
-    getHeaderGroupProps: any[]
-    getHeaderProps: any[]
-    getCellProps: any[]
+  export interface UseExpandedHooks<D extends object> {
+    getExpandedToggleProps: ((
+      row: Row<D>,
+      instance: TableInstance<D>
+    ) => object)[]
   }
 
-  export interface RowsProps {
-    subRowsKey: string
+  export interface UseExpandedState<D extends object> {
+    expanded: IdType<D>[]
   }
 
-  export interface FiltersProps {
-    filterFn: () => void
+  export interface UseExpandedInstanceProps<D extends object> {
+    rows: Row<D>[]
+  }
+
+  export interface UseExpandedRowProps<D extends object> {
+    isExpanded: boolean
+    toggleExpanded: (isExpanded?: boolean) => void
+  }
+  /* #endregion */
+
+  /* #region useFilters */
+  export function useFilters<D extends object = {}>(hooks: Hooks<D>): void
+
+  export type UseFiltersOptions<D extends object> = Partial<{
     manualFilters: boolean
     disableFilters: boolean
-    setFilter: () => any
-    setAllFilters: () => any
-  }
+    filterTypes: Filters<D>
+  }>
 
-  export interface UsePaginationState {
-    pageIndex: number
-    pageSize: number
-    pageCount: number
-    rowCount: number
+  export interface UseFiltersState<D extends object> {
+    filters: Filters<D>
   }
 
-  export interface UsePaginationValues<D = {}> {
-    page: Page<D>
-    pageIndex: number // yes, this is on instance and state
-    pageSize: number // yes, this is on instance and state
-    canPreviousPage: boolean
-    canNextPage: boolean
-    nextPage: () => any
-    previousPage: () => any
-    setPageSize: (size: number) => any
-    pageOptions: any[]
-    manualPagination: boolean
-    paginateExpandedRows: boolean
-    disablePageResetOnDataChange: boolean
-    pageCount: number
-    gotoPage: (page: number) => any
+  export type UseFiltersColumnOptions<D extends object> = Partial<{
+    Filter: ComponentType<TableInstance<D>> | ReactNode
+    disableFilters: boolean
+    filter: string | FilterType<D>
+  }>
+
+  export interface UseFiltersInstanceProps<D extends object> {
+    rows: Row<D>[]
+    preFilteredRows: Row<D>[]
+    setFilter: (
+      columnId: IdType<D>,
+      updater: FilterType<D> | ((filter: FilterType<D>) => FilterType<D>)
+    ) => void
+    setAllFilters: (
+      updater: Filters<D> | ((filters: Filters<D>) => Filters<D>)
+    ) => void
   }
 
-  export interface UseFiltersState<D> {
-    filters?: Filters<D>
+  export interface UseFiltersColumnProps<D extends object> {
+    canFilter: boolean
+    setFilter: (filterValue: FilterType<D>) => void
+    filterValue: FilterType<D>
+    preFilteredColumnRows: Row<D>[]
```

Comment:
Pretty sure that this is a bug in the docs, there's no reference to this in the docs, and the example code all uses `preFilteredRows`

Reply:
You're correct.

Reply:
Updated the docs as well